basiclti/basiclti-impl/src/java/org/sakaiproject/basiclti/impl/BasicLTISecurityServiceImpl.java

```diff
@@ -297,6 +297,40 @@ private void redirectOIDC(HttpServletRequest req, HttpServletResponse res,
 
 	}
 
+	/**
+	 * Handle the LTI 1.1.2 roud trip logic
+	*/
+	private void handleLTI112(HttpServletRequest req, HttpServletResponse res, Map<String, Object> tool)
+	{
+		String default_launch_type = ServerConfigurationService.getString(SakaiBLTIUtil.BASICLTI_LTI11_LAUNCH_TYPE, 
+				SakaiBLTIUtil.BASICLTI_LTI11_LAUNCH_TYPE_DEFAULT);
+		Long lti11_launch_type = SakaiBLTIUtil.getLongKey(tool.get(LTIService.LTI11_LAUNCH_TYPE));
+
+		if ( SakaiBLTIUtil.isLTI13(tool, null) ) return;
+
+		if ( lti11_launch_type.equals(LTIService.LTI11_LAUNCH_TYPE_LEGACY) ) return;
+
+		if ( lti11_launch_type.equals(LTIService.LTI11_LAUNCH_TYPE_LTI112) ||
+					SakaiBLTIUtil.BASICLTI_LTI11_LAUNCH_TYPE_LTI112.equals(default_launch_type) ) {
+
+			String tool_state = req.getParameter("tool_state");
+			if ( StringUtils.isEmpty(tool_state) ) {
+				// req.getRequestURL()=http://localhost:8080/access/basiclti/site/85fd092b-1755-4aa9-8abc-e6549527dce0/content:0
+				// req.getRequestURI()=/access/basiclti/site/85fd092b-1755-4aa9-8abc-e6549527dce0/content:0
+				String platform_state = req.getRequestURI();
+				String query_string = req.getQueryString();
+					if ( StringUtils.isNotEmpty(query_string)) {
```

Comment:
tabs/spacing looks off in this section

Reply:
Thanks - Took a bit to figure out it was just one line :)  I looking for spaces.